dev/bots/test.dart

```diff
@@ -471,11 +412,101 @@ Future<void> _runCoverage() async {
   );
   if (!coverageFile.existsSync()) {
     print('${red}Coverage file not found.$reset');
-    print('Expected to find: ${coverageFile.absolute}');
-    print('This file should have been generated by the `flutter test --coverage` script, but was not.');
+    print('Expected to find: $cyan${coverageFile.absolute}$reset');
+    print('This file should have been generated by the `${green}flutter test --coverage$reset` script, but was not.');
     exit(1);
   }
-  print('${bold}DONE: Coverage collection successful.$reset');
+}
+
+Future<void> _runWebTests() async {
+  final Map<String, ShardRunner> subshards = <String, ShardRunner>{};
+
+  final Directory flutterPackageDirectory = Directory(path.join(flutterRoot, 'packages', 'flutter'));
+  final Directory flutterPackageTestDirectory = Directory(path.join(flutterPackageDirectory.path, 'test'));
+
+  // TODO(yjbanov): we're getting rid of this blacklist as part of https://github.com/flutter/flutter/projects/60
+  const Set<String> kBlacklist = <String>{
+    'cupertino',
+    'examples',
+    'material',
+    'widgets',
+  };
+
+  final List<String> allTests = flutterPackageTestDirectory
+    .listSync()
+    .whereType<Directory>()
+    .where((Directory directory) => !kBlacklist.contains(path.basename(directory.path)))
+    .expand((Directory directory) => directory
+      .listSync(recursive: true)
+      .where((FileSystemEntity entity) => entity.path.endsWith('_test.dart'))
+    )
+    .whereType<File>()
+    .map<String>((File file) => path.relative(file.path, from: flutterPackageDirectory.path))
+    .toList()
+    ..shuffle(math.Random(0));
```

Comment:
Why shuffle it at all if we're going to hard code the seed?

Reply:
I'm not sure what Ian's motivation was, but I like this change because randomizing should break up clusters of big tests making sharding more effective.

Reply:
Might be worth a commend explaining why this is done.

Reply:
yeah you want to shuffle so that the average cost-per-file is uniform.
i'll add a comment.